tests/flows/test_sklearn.py

```diff
@@ -0,0 +1,459 @@
+        dummy = sklearn.dummy.DummyClassifier(strategy='prior')
+        model = sklearn.pipeline.Pipeline(steps=(
+            ('scaler', scaler), ('dummy', dummy)))
+
+        fixture_name = 'sklearn.pipeline.Pipeline(' \
+                       'sklearn.preprocessing.data.StandardScaler,sklearn.dummy.DummyClassifier)'
+        fixture_description = 'Automatically created sub-component.'
+
+        serialization =  sklearn_to_flow(model)
+
+        self.assertEqual(serialization.name, fixture_name)
+        self.assertEqual(serialization.description, fixture_description)
+
+        # Comparing the pipeline
+        # The parameters only have the name of base objects(not the whole flow)
+        # as value
+        self.assertEqual(len(serialization.parameters), 1)
+        # Hard to compare two representations of a dict due to possibly
+        # different sorting. Making a json makes it easier
+        self.assertEqual(json.loads(serialization.parameters['steps']),
+                         [{'oml-python:serialized_object':
+                               'component_reference', 'value': {'key': 'steps__scaler', 'step_name': 'scaler'}},
+                          {'oml-python:serialized_object':
+                               'component_reference', 'value': {'key': 'steps__dummy', 'step_name': 'dummy'}}])
+
+        # Checking the sub-component
+        self.assertEqual(len(serialization.components), 2)
+        self.assertIsInstance(serialization.components['steps__scaler'],
+                              OpenMLFlow)
+        self.assertIsInstance(serialization.components['steps__dummy'],
+                              OpenMLFlow)
+
+        #del serialization.model
+        new_model = flow_to_sklearn(serialization)
+
+        self.assertEqual(type(new_model), type(model))
+        self.assertIsNot(new_model, model)
+
+        self.assertEqual([step[0] for step in new_model.steps],
+                         [step[0] for step in model.steps])
+        self.assertIsNot(new_model.steps[0][1], model.steps[0][1])
+        self.assertIsNot(new_model.steps[1][1], model.steps[1][1])
+
+        new_model_params = new_model.get_params()
+        del new_model_params['scaler']
+        del new_model_params['dummy']
+        del new_model_params['steps']
+        fu_params = model.get_params()
+        del fu_params['scaler']
+        del fu_params['dummy']
+        del fu_params['steps']
+
+        self.assertEqual(new_model_params, fu_params)
+        new_model.fit(self.X, self.y)
+
+    def test_serialize_feature_union(self):
+        ohe = sklearn.preprocessing.OneHotEncoder(sparse=False)
+        scaler = sklearn.preprocessing.StandardScaler()
+        fu = sklearn.pipeline.FeatureUnion(transformer_list=[('ohe', ohe),
+                                                             ('scaler', scaler)])
+        serialization =  sklearn_to_flow(fu)
+        self.assertEqual(serialization.name,
+                         'sklearn.pipeline.FeatureUnion('
```

Comment:
Now it does. This is the same error as your last example which showed potential name collisions when in a feature union the steps were the same, but the names of two steps were switched, right?